### PR TITLE
Feature/107/inventory seo improvements

### PR DESCRIPTION
**Pull request by gluhovSemen**

### CODE REVIEW

File: src/apps/inventory/models.py [hunk 1]
```diff
@@ -1,14 +1,24 @@
 from django.conf import settings
 from django.db import models
+from django.utils.text import slugify
 
 
 class Category(models.Model):
     name = models.CharField(max_length=255)
     description = models.TextField(blank=True)
+    slug = models.SlugField(default=str(name))
```
Comment (egorvavilov) [inline src/apps/inventory/models.py]: Does it make sense? Feels like default arg should be omitted

File: src/apps/inventory/models.py [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
     def __str__(self):
         return self.name
 
+    def save(self, *args, **kwargs):
+        self.slug = slugify(self.name)
+        super().save(*args, **kwargs)
+
+    def get_absolute_url(self):
+        # Define the URL for a category
+        return f"/category/{self.slug}/"  # Adjust the URL structure as needed
```
Comment (egorvavilov) [inline src/apps/inventory/models.py]: why not reverse?

File: src/apps/inventory/tests.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+class ProductBySellerListViewTestCase(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(username="testuser2", password="password")
+        self.url = reverse("product_by_seller")
+
+    def test_authenticated_user_view(self):
+        self.client.login(username="testuser", password="password")
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "product_list.html")
```
Comment (egorvavilov) [inline src/apps/inventory/tests.py:103]: Test failed scenarios. What if category doesnt exist? what if product doesnt exist? what if product has different category

File: src/apps/inventory/migrations/0005_category_and_product_slug_populating.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    for category in Category.objects.all():
+        category.slug = f'{category.name}_{category.id}'
+        category.save()
+
+def reverse_populate_slug(apps, schema_editor):
+    Product = apps.get_model('inventory', 'Product')
+    Category = apps.get_model('inventory', 'Category')
+
+    for product in Product.objects.all():
+        product.slug = ''
```
Comment (egorvavilov) [inline src/apps/inventory/migrations/0005_category_and_product_slug_populating.py]: u have unique field and you manually set all records to ''

File: src/apps/inventory/models.py [hunk 5]
```diff
@@ -22,6 +32,14 @@ class Product(models.Model):
         limit_choices_to={"user_type": "Seller"},
     )
     created_at = models.DateTimeField(auto_now_add=True)
+    slug = models.SlugField(unique=True)
 
     def __str__(self):
         return self.name
+
+    def save(self, *args, **kwargs):
+        self.slug = slugify(self.name)
```
Comment (JedrzejMaluszczak) [inline src/apps/inventory/models.py]: be aware that if you ever update the name of a category or product, the slug won't automatically update. 
 Maybe we should consider using something like  https://pypi.org/project/django-autoslug/
@egorvavilov what do you think? I haven't use django for a while

File: src/apps/inventory/models.py [hunk 6]
```diff
@@ -22,6 +32,14 @@ class Product(models.Model):
         limit_choices_to={"user_type": "Seller"},
     )
     created_at = models.DateTimeField(auto_now_add=True)
+    slug = models.SlugField(unique=True)
```
Comment (JedrzejMaluszczak) [inline src/apps/inventory/models.py]: we should validate if Product.price has a positive value

File: src/apps/inventory/migrations/0005_category_and_product_slug_populating.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    Product = apps.get_model('inventory', 'Product')
+    Category = apps.get_model('inventory', 'Category')
+
+    for product in Product.objects.all():
+        product.slug = f'{product.name}_{product.id}'
+        product.save()
+
+    for category in Category.objects.all():
+        category.slug = f'{category.name}_{category.id}'
+        category.save()
```
Comment (JedrzejMaluszczak) [inline src/apps/inventory/migrations/0005_category_and_product_slug_populating.py]: you create a slug using name and id here but in save methods you still use only name

File: src/apps/inventory/models.py (hunk 5 above)
Comment (egorvavilov) [reply src/apps/inventory/models.py]: exactly, thank you for noticing!

File: src/apps/inventory/models.py (hunk 6 above)
Comment (egorvavilov) [reply src/apps/inventory/models.py]: lets keep it as it is now, we will build business requirements bit later

File: src/apps/inventory/migrations/0005_category_and_product_slug_populating.py (hunk 4 above)
Comment (gluhovSemen) [reply src/apps/inventory/migrations/0005_category_and_product_slug_populating.py]: I dont see issue here i first create a filed without unic in 0004 migration then populate it and then add restriction, and do the reveres in the opposite order

File: src/apps/inventory/models.py (hunk 1 above)
Comment (gluhovSemen) [reply src/apps/inventory/models.py]: fixed

File: src/apps/inventory/models.py (hunk 2 above)
Comment (gluhovSemen) [reply src/apps/inventory/models.py]: fixed

File: src/apps/inventory/tests.py (hunk 3 above)
Comment (gluhovSemen) [reply src/apps/inventory/tests.py:103]: fixed

File: src/apps/inventory/models.py (hunk 5 above)
Comment (gluhovSemen) [reply src/apps/inventory/models.py]: done

File: src/apps/inventory/migrations/0005_category_and_product_slug_populating.py (hunk 7 above)
Comment (gluhovSemen) [reply src/apps/inventory/migrations/0005_category_and_product_slug_populating.py]: fixed

**Review by egorvavilov — APPROVED** [review]

**Pull request merged**